parser/parser.y

```diff
@@ -1798,30 +1811,74 @@ PrimaryExpression:
 	}
 
 Function:
-	PrimaryExpression FunctionCall 
+	FunctionCallKeyword
+|	FunctionCallNonKeyword
+|	FunctionCallConflict
+
+FunctionCallArgList:
+	/* EMPTY */
 	{
-		x := yylex.(*lexer)
-		f, ok := $1.(*expressions.Ident)
-		if !ok {
-			x.err("", "expected identifier or qualified identifier")
-			return 1
-		}
+		$$ = []expression.Expression{}
+	}
+	/* select count(*) from table */
+|	'*'  
+	{
+		$$ = []expression.Expression{ expressions.Value{Val: expressions.TypeStar("*")} }
+	}
+|	ExpressionList
+
+FunctionCall:
+	'(' AggAllOpt FunctionCallArgList ')'
+	{
+		$$ = []interface{}{false, $3}
+	}
+|	'(' "DISTINCT" AggAllOpt ExpressionList')'
```

Comment:
only aggregate function needs distinct, should we remove it here?

but I don't know whether other functions use "all".